build-qt.sh

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env bash
+set -e
+set -u
+set -o pipefail
+
+prefix_output() {
+    prefix="${1}"
+    shift
+    "$@" 2> >(sed "s/^/${prefix}/" >&2) | sed "s/^/${prefix}/"
+}
+
+module_vars() {
+    url="https://download.qt.io/archive/qt/6.8/6.8.3/submodules/${1}-everywhere-src-6.8.3.tar.xz"
+    archive="${url##*/}"
+    dirname="${archive%.tar.*}"
+}
+
+
+build_module() {
+    module="${1}"
+    module_vars "${module}"
+    cd "${WORK_DIR}/${dirname}"
+    options_var="options_${module}"
+    truncate_var="truncate_${module}"
+    echo ${!truncate_var:-} | xargs --no-run-if-empty truncate -s 0
+    if [[ "${module}" = qtbase ]]; then
+        ./configure ${!options_var} ${common_options_cmake}
+    else
+        "${INSTALL_DIR}/bin/qt-configure-module" . ${!options_var} ${common_options_cmake}
+    fi
+    ninja
+    cmake --install .
+}
+
+WORK_DIR="${PWD}"
+SCRIPT_DIR=$(dirname "$0")
+INSTALL_DIR="${WORK_DIR}/qt"
+
+#FIXME bad warning in qtdeclarative-everywhere-src-6.8.3/src/quickwidgets/qquickwidget.cpp
+
+common_options_cmake='-DQT_GENERATE_SBOM=OFF -DFEATURE_accessibility=OFF'
+options_qtbase="
+    -opensource
+    -confirm-license
+    -release
+    -optimize-size
+    -no-shared
+    -static
+    --c++std=17
+    -disable-deprecated-up-to 0x060800
+    -qpa xcb
+    -nomake tests
+    -nomake examples
+    -no-gif
+    -no-icu
+    -no-glib
+    -opengl desktop
+    -no-eglfs
+    -no-opengles3
+    -no-egl
+    -xkbcommon
+    -dbus-runtime
+    -qt-freetype
+    -qt-pcre
+    -qt-harfbuzz
+    -qt-libpng
+    -qt-libjpeg
+    -system-zlib
+    -openssl-linked
+    -prefix ${INSTALL_DIR}
+    --
+    -DFEATURE_androiddeployqt=OFF
+    -DFEATURE_animation=ON
+    -DFEATURE_cborstreamwriter=OFF
+    -DFEATURE_columnview=OFF
+    -DFEATURE_commandlinkbutton=OFF
+    -DFEATURE_completer=OFF
+    -DFEATURE_concatenatetablesproxy=OFF
+    -DFEATURE_concurrent=OFF
+    -DFEATURE_contextmenu=OFF
+    -DFEATURE_cssparser=OFF
+    -DFEATURE_datawidgetmapper=OFF
+    -DFEATURE_datetimeedit=OFF
+    -DFEATURE_dial=OFF
+    -DFEATURE_dialogbuttonbox=ON
+    -DFEATURE_dockwidget=OFF
+    -DFEATURE_draganddrop=OFF
+    -DFEATURE_dtls=OFF
+    -DFEATURE_easingcurve=ON
+    -DFEATURE_filedialog=OFF
+    -DFEATURE_filesystemmodel=OFF
+    -DFEATURE_filesystemwatcher=OFF
+    -DFEATURE_fontcombobox=OFF
+    -DFEATURE_fontdialog=OFF
+    -DFEATURE_formlayout=OFF
+    -DFEATURE_fscompleter=OFF
+    -DFEATURE_future=OFF
+    -DFEATURE_gc_binaries=OFF
```

Comment:
If this is the same as `-gc-binaries`, it should be `ON` to reduce size:

According to <https://doc.qt.io/qt-6/build-optimized-qt.html>:

> -gc-binaries
> Remove unnecessary parts from binary. Place each function or data item into its own section and enable linker garbage collection of unused sections.